### PR TITLE
Add Histogram and Heatmap visualization

### PR DESCRIPTION
**Pull request by ronensc**
This PR is on top of #152. Should be merged after that one is merged.

Solves issues: #19 #9 

Print screens of the added visualizations are attached below:
Heatmap in the details dashboard
image
Histogram in the totals dashboard
image

The visualizations are based on:
https://grafana.com/blog/2020/06/23/how-to-visualize-prometheus-histograms-in-grafana/

### CODE REVIEW

File: network_definitions/mice_elephants_histogram.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#flp_confgen
+description:
+  A histogram of flowlog bytes
+details:
+  A histogram of flowlog bytes
```
Comment (eranra) [inline network_definitions/mice_elephants_histogram.yaml:5]:
```suggestion
  Flows length distribution over time
```

File: network_definitions/mice_elephants_histogram.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+usage:
+  Evaluate network behaviour
```
Comment (eranra) [inline network_definitions/mice_elephants_histogram.yaml:7]:
```suggestion
  Evaluate flows length behaviour
```

File: network_definitions/mice_elephants_histogram.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  - rate
+transform:
+  rules:
+    - input: bytes
+      output: all
+      type: add_if
+      parameters: ">=0"
+extract:
+  aggregates:
+    - name: bytes_hist
```
Comment (eranra) [inline network_definitions/mice_elephants_histogram.yaml:21]: s/bytes_hist/flows_bytes_hist

**Review by eranra — APPROVED** [review]
This looks very good to me ... all the suggestions are updates for names, variables etc ... in general we do not need to see any more mice and elephants anywhere ... everything can move to be  "flows length"

File: contrib/kubernetes/flowlogs-pipeline.conf.yaml [hunk 4]
```diff
@@ -303,6 +312,21 @@ parameters:
...
+        type: histogram
+        filter:
+          key: name
+          value: bytes_hist
+        valuekey: recent_raw_values
+        labels:
+        - by
+        - aggregate
+        buckets:
+        - 100
```
Comment (eranra) [inline contrib/kubernetes/flowlogs-pipeline.conf.yaml:325]: @ronensc can you update to 128,1024 etc ...

File: docs/metrics.md [hunk 5]
```diff
@@ -145,6 +145,19 @@ and the transformation to generate the exported metric.
 |||  
 
 
+### flows length histogram
+| **Description** | A histogram of flowlog bytes | 
+|:---|:---|
+| **Details** | Flows length distribution over time | 
+| **Usage** | Evaluate flows length behavior including mice/elephant use-case |
```
Comment (KalmanMeth) [inline docs/metrics.md:152]: is the mention of "mice/elephants" here (and below) still intended?

Comment (ronensc) [reply docs/metrics.md:152]: Eran has updated it recently so I guess it is intended 
https://github.com/netobserv/flowlogs-pipeline/pull/164#discussion_r841189736

Comment (eranra) [reply docs/metrics.md:152]: @KalmanMeth this is the only place where we explain that the histogram is useful for the mice/elephant use-case --- hope that this make sense to you

**Pull request merged**